proofs/tree.go

```diff
@@ -470,26 +637,93 @@ func FlattenMessage(message, messageSalts proto.Message) (nodes [][]byte, propOr
 	return f.nodes, f.propOrder, nil
 }
 
-// getStringValueByProperty gets a value from a struct and returns the value. This method does not yet
-// support nested structs. It converts the value to a string representation.
+// normalizeDottedProperty performs camel case conversion without removing slice characters ([])
+func normalizeDottedProperty(prop string) (string) {
+	arr := strings.Split(prop, ".")
+	for idx, key := range arr {
+		var propAux string
+		re := regexp.MustCompile(`(.*)(\[(.*)])`)
+		if re.MatchString(key) {
+			prefix := re.ReplaceAllString(key, "$1")
+			strIdx := re.ReplaceAllString(key, "$3")
+			propAux = fmt.Sprintf("%s[%s]", strcase.ToCamel(prefix), strIdx)
+		} else {
+			propAux = strcase.ToCamel(key)
+		}
+
+		if idx == 0 {
+			prop = fmt.Sprintf("%s", propAux)
+		} else {
+			prop = fmt.Sprintf("%s.%s", prop, propAux)
+		}
+	}
+	return prop
+}
+
+// getDottedValueByProperty takes a dotted property and retrieves the interface value of an object
+// It supports nested fields and slices
+func getDottedValueByProperty(prop string, value interface{}) (interface{}, error) {
+	prop = normalizeDottedProperty(prop)
+	if reflect.TypeOf(value).Kind() != reflect.Struct && reflect.TypeOf(value).Kind() != reflect.Ptr {
+		return nil, errors.New("non-struct interface not supported")
+	}
+	arr := strings.Split(prop, ".")
+	value = DereferencePointer(value)
+	var err error
+	for _, key := range arr {
+		re := regexp.MustCompile(`(.*)(\[(.*)])`)
+		if re.MatchString(key) {
+			prefix := re.ReplaceAllString(key, "$1")
+			strIdx := re.ReplaceAllString(key, "$3")
+			idx, err := strconv.Atoi(strIdx)
+			if err != nil {
+				return nil, err
+			}
+			value, err = reflections.GetField(value, prefix)
+			if err != nil {
+				return nil, err
+			}
+			if value == nil {
+				return nil, nil
+			}
```

Comment:
Can we return an error here as well ?

Reply:
the value of a field can be nil, that doesnt mean that is an error. The `reflections.GetField` returns an error if the field doesnt exist.

Reply:
Then, let's ensure we mention that `nil, nil` is possible since the usual assumption would be that is `err == nil` then other value is not

Reply:
Ok I will add some documentation about it.